erts/emulator/nifs/common/prim_tty_nif.c

```diff
@@ -142,14 +139,12 @@ static ERL_NIF_TERM tty_tgetnum_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM
 static ERL_NIF_TERM tty_tgetflag_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM tty_tgetstr_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 static ERL_NIF_TERM tty_tgoto_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
-static ERL_NIF_TERM tty_read_signal_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]);
 
 static ErlNifFunc nif_funcs[] = {
     {"isatty", 1, isatty_nif},
     {"tty_create", 0, tty_create_nif},
     {"tty_init", 3, tty_init_nif},
     {"tty_set", 1, tty_set_nif},
-    {"tty_read_signal", 2, tty_read_signal_nif},
     {"setlocale", 1, setlocale_nif},
     {"tty_select", 3, tty_select_nif},
```

Comment:
> I submitted the patch with tty_select/3 unused to share the patch and start the discussion.

Change this to `{"tty_select", 3, tty_select_nif},` and in that function, change `argv[2]` to `argv[1]` to account for the removed argument in the middle. :-)

Reply:
Sorry, I noticed I hadn't updated it to `2` a few seconds after posting. Guess you missed my edit :-(

`{"tty_select", 3, tty_select_nif},` means that `tty_select/3` has to be present in the module for NIF loading to work. If you want `tty_select/2`, change it to `{"tty_select", 2, tty_select_nif},`.

Reply:
That’s what I did yesterday but it fails to compilet with the error pasted above.

I see that `prim_tty.beam` is present in the bootstrap. I looked at `HOWTO/BOOTSTRAP.md` but it says that it shouldn’t be modified during development.

Reply:
Ah yes. To make this work while testing, run `./otp_build update_bootstrap --no-commit`.

Reply:
With `./otp_build update_primary --no-commit` and an improvement to take into account that `erl_signal_server` may not be running, I can compile locally with `tty_select/2`. The `interactive_shell_SUITE` is still happy.

However, CI is still failing. Should I commit the modules in `bootstrap`?

Reply:
> However, CI is still failing. Should I commit the modules in bootstrap?

That won't be necessary, I'll add it manually to our CI with updated bootstrap. :-)

Reply:
Ok. I don’t have any planned changes left, but I don’t know what to do with CI :-) Do you want me to mark the pull request as ready for review?

Reply:
> With `./otp_build update_primary --no-commit` and an improvement to take into account that `erl_signal_server` may not be running, I can compile locally with `tty_select/2`. The `interactive_shell_SUITE` is still happy.

That's odd, `erl_signal_server` should've been started before `user` (and thereby `user_sup` -> `user_drv` -> `prim_tty`), and I'm not sure that ignoring its absence is the right thing to do. @garazdawi ?

Reply:
> Ok. I don’t have any planned changes left, but I don’t know what to do with CI :-) Do you want me to mark the pull request as ready for review?

Sure :-)

Reply:
> That's odd, unless I'm missing something `erl_signal_server` should've been started before `user`

That’s how I interpret the compile-time error: `gen_event:delete_handler/3` throws a `noproc` error. Perhaps this is a specific situation during the compilation if `prim_tty` from bootstrap is used?

Because with the `interactive_shell_SUITE` or manual testing with an Erlang shell, I didn’t hit this error.

Reply:
The signal server is not started when `-mode minimal` is passed, so that is why it fails with `erlc` as it uses that flag. I pushed a commit that moves when it is started so that it is always present.